### PR TITLE
Added expires_at parameter

### PR DESCRIPTION
**Pull request by AnasNaouchi**

### CODE REVIEW

File: Omise/Models/Models.cs [hunk 1]
```diff
@@ -336,6 +336,8 @@ public partial class Charge : ModelBase {
         public string Currency { get; set; }
         [JsonProperty("description")]
         public string Description { get; set; }
+        [JsonProperty("expires_at")]
+        public string ExpiresAt { get; set; }
```
Comment (necessarylion) [inline Omise/Models/Models.cs:340]: datatype should be `DateTime` same as `paid_at`.

Comment (ajzkk) [inline Omise/Models/Models.cs]: datatype should be `DateTime` same as `paid_at`?

File: Omise.Examples/ExampleInfo.cs [hunk 2]
```diff
@@ -1,36 +1,35 @@
-﻿
-namespace Omise.Examples
+﻿namespace Omise.Examples
 {
     public class ExampleInfo
     {
-        public const string OMISE_PKEY = "pkey_test_5alr4tca4bpu0cirhck";
-        public const string OMISE_SKEY = "skey_test_5alr4tz3so072nhkipj";
+        public const string OMISE_PKEY = "pkey_test_5uhkt2eb9s1xtbbp4v1";
+        public const string OMISE_SKEY = "skey_test_5uhkt2fyyjdhork2sjh";
```
Comment (aashishgurung) [inline Omise.Examples/ExampleInfo.cs]: Can you please help me understand why do we need active keys here?

Comment (AnasNaouchi) [reply Omise.Examples/ExampleInfo.cs]: Because the example app uses real api calls although I would say its better to indicate to the user that he should replace these parameters with his own parameters.

File: Omise/Models/Models.cs (hunk 1 above)
Comment (AnasNaouchi) [reply Omise/Models/Models.cs]: Yeah it should be will change

File: Omise.Examples/ExampleInfo.cs (hunk 2 above)
Comment (aashishgurung) [reply Omise.Examples/ExampleInfo.cs]: Can we handle this via env variables with the help of library like [dotenv.net](https://github.com/bolorundurowb/dotenv.net)? So that we don't have to put the real keys here.

File: Omise.Examples/Examples/Charges.cs [hunk 3]
```diff
@@ -49,6 +49,7 @@ public async Task Create__Create_With_Card()
                 Currency = "thb",
                 Customer = customerId,
                 Card = cardId,
+                ExpiresAt= "2023-08-10T23:01:50Z"
```
Comment (danfowler) [inline Omise.Examples/Examples/Charges.cs]:
```suggestion
                ExpiresAt = "2023-08-10T23:01:50Z"
```

File: Omise.Examples/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -17,7 +17,7 @@
 // The form "{Major}.{Minor}.*" will automatically update the build and revision,
 // and "{Major}.{Minor}.{Build}.*" will update just the revision.
 
-[assembly: AssemblyVersion("1.0.*")]
+[assembly: AssemblyVersion("1.0.0.0")] // no wildcard to be compatible with determinism
```
Comment (danfowler) [inline Omise.Examples/Properties/AssemblyInfo.cs:20]: https://en.wikipedia.org/wiki/Determinism?

File: Omise.Tests/Resources/ChargeResourceTest.cs [hunk 5]
```diff
@@ -2,66 +2,65 @@
...
 {
     [TestFixture]
-    public class ChargeResourceTest : ResourceTest<ChargeResource>
+    public class ChargeResourceTest : ResourceTest<ChargeResource>
     {
         const string ChargeId = "chrg_test_4yq7duw15p9hdrjp8oq";
 
         [Test]
-        public async Task TestGetList()
+        public async Task TestGetList()
```
Comment (danfowler) [inline Omise.Tests/Resources/ChargeResourceTest.cs]: It seems like there's no actual change here so can we exclude these false changes?

File: Omise.Examples/Properties/AssemblyInfo.cs (hunk 4 above)
Comment (AnasNaouchi) [reply Omise.Examples/Properties/AssemblyInfo.cs:20]: @danfowler https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/code-generation#deterministic

File: Makefile [hunk 6]
```diff
@@ -15,7 +15,7 @@ T4_OUTPUT_FILES := $(T4_FILES:.tt=.cs)
 
 MONO    := mono
 MSBUILD := msbuild /p:Configuration=$(CONFIG)
-NUNIT := $(MONO) packages/NUnit.ConsoleRunner.3.9.0/tools/nunit3-console.exe
+NUNIT := dotnet test
```
Comment (AnasNaouchi) [inline Makefile:18]: Moved to standard dotnet testing tool allowing us to run tests using multiple testing frameworks (like NUnit, xUnit, MSTest) and not just NUnit like what was before. Also the testing was failing as the nunit3-console.exe file was not found in the specified path but this change allowed us to remove third party dependency.

File: Omise.Tests/Resources/ChargeResourceTest.cs [hunk 7]
```diff
@@ -149,9 +151,12 @@ public async Task TestFixturesSearch()
             Assert.That(result[0].Amount, Is.EqualTo(409669));
         }
 
-        protected CreateChargeRequest BuildCreateRequest()
+        protected CreateChargeRequest BuildCreateRequest()
         {
-            return new CreateChargeRequest
+            TimeZoneInfo thailandZone = TimeZoneInfo.FindSystemTimeZoneById("Asia/Bangkok");
+            DateTime thailandTime = new DateTime(2023, 8, 9, 0, 0, 0);
+            DateTime utcTime = TimeZoneInfo.ConvertTimeToUtc(thailandTime, thailandZone);
```
Comment (AnasNaouchi) [inline Omise.Tests/Resources/ChargeResourceTest.cs:158]: Fix the timezone across multi working environments so that tests don't fail from a machine to another (locally or on github actions server as utc is the default on server but locally depends on your timezone)

File: Omise.Tests/packages.config [hunk 8]
```diff
@@ -6,7 +6,6 @@
   <package id="Newtonsoft.Json" version="13.0.2" targetFramework="net462" />
   <package id="NuGet.Build.Packaging" version="0.2.2" targetFramework="net462" developmentDependency="true" />
   <package id="NUnit" version="3.11.0" targetFramework="net462" />
-  <package id="NUnit.ConsoleRunner" version="3.9.0" targetFramework="net462" />
```
Comment (AnasNaouchi) [inline Omise.Tests/packages.config:9]: Removed legacy package reference as this framework is not used anymore

File: Omise.Tests/Resources/ChargeResourceTest.cs [hunk 9]
```diff
@@ -83,6 +84,7 @@ public void TestCreateChargeRequest()
                 @"""amount"":244884," +
                 @"""currency"":""thb""," +
                 @"""description"":""Test Charge""," +
+                @"""expires_at"":""2023-08-08T17:00:00Z""," +
```
Comment (annatotest) [inline Omise.Tests/Resources/ChargeResourceTest.cs:87]: Can we make this test to do not fail in the future?

Comment (AnasNaouchi) [reply Omise.Tests/Resources/ChargeResourceTest.cs:87]: it will not fail, as the test is fixed on this date

File: Omise/Omise.csproj [hunk 10]
```diff
@@ -3,7 +3,7 @@
   <PropertyGroup>
     <TargetFramework>netstandard2.0</TargetFramework>
     <PackageId>Omise</PackageId>
-    <PackageVersion>2.7.14</PackageVersion>
+    <PackageVersion>2.7.8</PackageVersion>
```
Comment (ajzkk) [inline Omise/Omise.csproj]: Why are we bumping to lower version? Do you mean `2.8.0`?
And I think it should do with separate PR right? Example [here](https://github.com/omise/omise-dotnet/commit/cf446d6454c6b7246bbe51a7afe18da293057cdc).

cc @danfowler

Comment (AnasNaouchi) [reply Omise/Omise.csproj]: Yeah I meant 2.8.0, should we really do a separate PR?

File: Omise.Examples/Properties/AssemblyInfo.cs (hunk 4 above)
Comment (danfowler) [reply Omise.Examples/Properties/AssemblyInfo.cs:20]: 👍🏽

File: Omise/Omise.csproj (hunk 10 above)
Comment (danfowler) [reply Omise/Omise.csproj]: Yes, let's bump in a separate PR. Also, yes, the version is very wrong.

Comment (AnasNaouchi) [reply Omise/Omise.csproj]: ok

File: Omise.Tests/Resources/ChargeResourceTest.cs (hunk 5 above)
Comment (danfowler) [reply Omise.Tests/Resources/ChargeResourceTest.cs]: @AnasNaouchi why do these lines show a change when there is no actual change?

File: Omise/Omise.csproj (hunk 10 above)
Comment (AnasNaouchi) [reply Omise/Omise.csproj]: Reversed the Bump, will do a separate PR for release once test cases are ready from @annatotest

File: Omise.Tests/Resources/ChargeResourceTest.cs (hunk 5 above)
Comment (AnasNaouchi) [reply Omise.Tests/Resources/ChargeResourceTest.cs]: There is no change but github keeps showing a change

Comment (ajzkk) [reply Omise.Tests/Resources/ChargeResourceTest.cs]: I think it is end of line sequence (LF vs CRLF) issue. You can change it in vscode footer. May be you switch to different one by mistake.

File: Omise/Models/Models.cs [hunk 11]
```diff
@@ -1,6 +1,6 @@
-﻿// <auto-generated />
+// <auto-generated />
 // Generated from /Omise/Models/Models.tt
-// on 09-Aug-22 5:56:24 PM
+// on 10/8/2566 พ.ศ. 12:18:32
```
Comment (danfowler) [inline Omise/Models/Models.cs:3]: @AnasNaouchi why is this in Thai format?

Comment (AnasNaouchi) [reply Omise/Models/Models.cs:3]: It is auto generated

File: Omise.Tests/Resources/ChargeResourceTest.cs (hunk 5 above)
Comment (danfowler) [reply Omise.Tests/Resources/ChargeResourceTest.cs]:
```
git config diff.wsErrorHighlight all
git diff
```

File: Omise/Models/Models.cs (hunk 11 above)
Comment (annatotest) [reply Omise/Models/Models.cs:3]: What is it used for?

Comment (AnasNaouchi) [reply Omise/Models/Models.cs:3]: Just an indicator for the time at which the models were generated

Comment (danfowler) [reply Omise/Models/Models.cs:3]: @AnasNaouchi I would recommend figuring out how to ensure that the date string is created in a consistent format and add that to the repo

Comment (danfowler) [reply Omise/Models/Models.cs:3]: @AnasNaouchi what command do you run to generate the Models.cs file?

File: Omise/Models/ChargeRequest.cs [hunk 12]
```diff
@@ -1,4 +1,5 @@
 ﻿using Newtonsoft.Json;
+using System;
```
Comment (danfowler) [inline Omise/Models/ChargeRequest.cs:2]: @AnasNaouchi is there a reason why `using System;` is needed now when it wasn't before?

Comment (AnasNaouchi) [reply Omise/Models/ChargeRequest.cs:2]: yes since we are using `DateTime` type from the namsespace `System` to set the type for `ExpiresAt` parameter

**Review by danfowler — APPROVED** [review]

**Review by ajzkk — APPROVED** [review]
